RoslynPluginGenerator/DiagnosticAssemblyScanner.cs

```diff
@@ -36,12 +36,31 @@ public DiagnosticAssemblyScanner(ILogger logger, params string[] additionalSearc
         /// <returns>Enumerable with instances of DiagnosticAnalyzer from discovered assemblies</returns>
         public IEnumerable<DiagnosticAnalyzer> InstantiateDiagnostics(string language, params string[] files)
         {
-            List<DiagnosticAnalyzer> analyzers = new List<DiagnosticAnalyzer>();
-            foreach (string assemblyPath in files.Where(f => Utilities.IsAssemblyLibraryFileName(f)))
+            // If there were any additional assembly search directories specified in the constructor, use them
```

Comment:
I moved the assembly resolver code so there is a single assembly resolver instance that covers the whole process. Previously, a separate resolver was created for each assembly, and only for the load part i.e. it was disposed before the types in the assembly were fetched.